### PR TITLE
Wildcard V2: <Icon /> Codemod

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
## Description
Implement Codemod tool, to automate migration usage of `icon-inline` className to `<Icon/>` wildcard component.

## Refs
[SG Issue](https://github.com/sourcegraph/sourcegraph/issues/27669)
[Gitstart Ticket](https://app.gitstart.com/clients/sourcegraph/tickets/SG-27669)

### CODE REVIEW

**Review by valerybugakov — CHANGES_REQUESTED** [review]
1. I think all props should be passed to the `<Icon />`  component. I noticed the current `<Icon />` implementation differs from the API described in [the original issue](https://github.com/sourcegraph/sourcegraph/issues/27667):

```tsx
// Current implementation
<Icon svg={<CloseIcon />} />

// VS desired 
<Icon svg={CloseIcon} />
```

Changing the API that way should solve the issue because all props will be consumed the SVG element in the internal implementation.

2. Could the missing check like that result in issues? I'm not sure I understood the question. Please provide an example if it's still relevant.

I [pushed the codemod update](https://github.com/sourcegraph/codemod/pull/87/commits/679fddd4695ad918eacd5bd0264d4ab2513ad6b4) that uses the API I mentioned in test cases, and it allows simplifying the codemod logic. Let me know if it makes sense, and let's not delete it 😃

Next steps:
1. Let's update the `Icon` component in the Sourcegraph codebase. 
2. Test the codemod locally against a couple of modules to see that it works. 
3. Update the codemod script if needed and request review again to merge it.
4. Run the codemod against Sourcegraph codebase to complete the `Icon` migration.

File: packages/transforms/src/iconToComponent/README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Icon element to `<Icon />` Wildcard component codemod
+
+yarn transform --write --tagToConvert=Link -t ./packages/transforms/src/iconToComponent/iconToComponent.ts '/Users/raphael/Documents/Development/work/Gitstart/client-sourcegraph/client/!(wildcard)/src/\*_/_.{tsx}'
```
Comment (valerybugakov) [inline packages/transforms/src/iconToComponent/README.md:3]: Before requesting a review, let's update this command by removing user-specific paths.

**Comment by gitstart-sourcegraph** [comment]
@valerybugakov Updating `Icon` implement would help to make this issue less complicated 😄 😄 😄  Thanks, update this soon

**Review by valerybugakov — COMMENTED** [review]
Hey @gitstart-sourcegraph, it seems that we've updated the `Icon` codemod locally to run against the Sourcegraph codebase. Does that mean we can update this PR?

File: packages/transforms/src/iconToComponent/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Icon element to `<Icon />` Wildcard component codemod
+
+yarn transform --write --tagToConvert=Link -t ./packages/transforms/src/iconToComponent/iconToComponent.ts '/sourcegraph/client/!(wildcard)/src/\*_/_.{ts,tsx}'
```
Comment (valerybugakov) [inline packages/transforms/src/iconToComponent/README.md:3]:
```suggestion
yarn transform --write -t ./packages/transforms/src/iconToComponent/iconToComponent.ts '/sourcegraph/client/!(wildcard)/src/\*_/_.{ts,tsx}'
```

File: packages/transforms/src/iconToComponent/iconToComponent.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import { validateCodemodTarget, validateCodemodTargetOrThrow } from './validateCodemodTarget'
+
+interface IconToComponentOptions {
+    tagToConvert: string
+}
+
+/**
+ * Convert `<SomeIcon class="icon-inline" />` element to the `<Icon svg={<SomeIcon />} />` component.
+ */
+export const iconToComponent = runTransform<IconToComponentOptions>(context => {
```
Comment (valerybugakov) [inline packages/transforms/src/iconToComponent/iconToComponent.ts:25]:
```suggestion

/**
 * Convert `<SomeIcon class="icon-inline" />` element to the `<Icon svg={<SomeIcon />} />` component.
 */
export const iconToComponent = runTransform(context => {
```

**Review by valerybugakov — APPROVED** [review]
Let's merge after removing the redundant transform option.

**Pull request merged**